src/main/kotlin/com/terraformation/backend/customer/db/UserStore.kt

```diff
@@ -178,6 +179,26 @@ class UserStore(
     return user?.let { rowToIndividualUser(it) }
   }
 
+  /**
+   * Returns the row for the user with a given email address.
+   *
+   * @param email Email address to search for. If the address has upper-case characters, they will
+   *   be folded to lower case.
+   * @return null if no user has the requested email address, or if users that do have been deleted.
+   */
+  fun fetchUserRowByEmail(email: String): UsersRow? {
```

Comment:
Would it make sense to make this return `TerrawareUser?` instead? That is, call `rowToModel()` on the row before returning it? That way callers wouldn't have to do null-safe field accesses on things like the user ID, and also it'd keep the public API of this class a little more consistent in that all the methods would return instances of `TerrawareUser` or one of its subclasses.

Reply:
It was mainly to avoid having to implement `rowToSystemUser` as well as avoid having to update the method anytime we add a new user type, but I'm fine that if it's preferred